src/formats/jpeg/frame_header.zig

```diff
@@ -0,0 +1,139 @@
+//! this module implements the frame header followint the t-81 specs,
+//! section b.2.2 Frame Header Syntax
+
+const std = @import("std");
+
+const buffered_stream_source = @import("../../buffered_stream_source.zig");
+const Image = @import("../../Image.zig");
+const ImageReadError = Image.ReadError;
+
+const Allocator = std.mem.Allocator;
+
+const JPEG_DEBUG = false;
+
+const Component = struct {
+    id: u8,
+    horizontal_sampling_factor: u4,
+    vertical_sampling_factor: u4,
+    quantization_table_id: u8,
+
+    pub fn read(reader: buffered_stream_source.DefaultBufferedStreamSourceReader.Reader) ImageReadError!Component {
+        const component_id = try reader.readByte();
+        const sampling_factors = try reader.readByte();
+        const quantization_table_id = try reader.readByte();
+
+        const horizontal_sampling_factor: u4 = @intCast(sampling_factors >> 4);
+        const vertical_sampling_factor: u4 = @intCast(sampling_factors & 0xF);
+
+        if (horizontal_sampling_factor < 1 or horizontal_sampling_factor > 4) {
+            // TODO(angelo): error, create cusotm error
+            return ImageReadError.InvalidData;
+        }
+
+        if (vertical_sampling_factor < 1 or vertical_sampling_factor > 4) {
+            // TODO(angelo): error, create custom error
+            return ImageReadError.InvalidData;
+        }
+
+        if (quantization_table_id > 3) {
+            // TODO(angelo): error, create custom error
+            return ImageReadError.InvalidData;
+        }
+
+        return Component{
+            .id = component_id,
+            .horizontal_sampling_factor = horizontal_sampling_factor,
+            .vertical_sampling_factor = vertical_sampling_factor,
+            .quantization_table_id = quantization_table_id,
+        };
+    }
+};
+
+const Self = @This();
```

Comment:
Not required, but it would be nice to have a extra line after this so it is not merged with the fields declaration.